src/panels/lovelace/editor/header-footer-editor/hui-dialog-create-headerfooter.ts

```diff
@@ -0,0 +1,193 @@
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { until } from "lit-html/directives/until";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { createCloseHeading } from "../../../../components/ha-dialog";
+import { HassDialog } from "../../../../dialogs/make-dialog-manager";
+import { haStyleDialog } from "../../../../resources/styles";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceHeaderFooterConfig } from "../../header-footer/types";
+import { headerFooterElements } from "../lovelace-headerfooters";
+import { HeaderFooter } from "../types";
+import { getHeaderFooterStubConfig } from "./get-headerfooter-stub-config";
+import { CreateHeaderFooterDialogParams } from "./show-create-headerfooter-dialog";
+
+@customElement("hui-dialog-create-headerfooter")
+export class HuiCreateDialogHeaderFooter extends LitElement
+  implements HassDialog<CreateHeaderFooterDialogParams> {
+  @property({ attribute: false }) protected hass!: HomeAssistant;
+
+  @internalProperty() private _params?: CreateHeaderFooterDialogParams;
+
+  public async showDialog(
+    params: CreateHeaderFooterDialogParams
+  ): Promise<void> {
+    this._params = params;
+  }
+
+  public closeDialog(): boolean {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+    return true;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        .heading=${createCloseHeading(
+          this.hass,
+          this.hass!.localize(
+            `ui.panel.lovelace.editor.header-footer.choose_header_footer`,
+            "type",
+            this.hass!.localize(
+              `ui.panel.lovelace.editor.header-footer.${this._params.type}`
+            )
+          )
+        )}
+        @keydown=${this._ignoreKeydown}
+        @closed=${this._cancel}
+      >
+        <div class="elements">
+          ${headerFooterElements.map((headerFooter) =>
+            until(
+              this._renderHeaderFooterElement(headerFooter),
+              html`
+                <div class="spinner">
+                  <ha-circular-progress
+                    active
+                    alt="Loading"
+                  ></ha-circular-progress>
+                </div>
+              `
+            )
+          )}
+        </div>
+        <div slot="primaryAction">
+          <mwc-button @click=${this._cancel}>
+            ${this.hass!.localize("ui.common.cancel")}
+          </mwc-button>
+        </div>
+      </ha-dialog>
+    `;
+  }
+
+  private async _renderHeaderFooterElement(
+    headerFooter: HeaderFooter
+  ): Promise<TemplateResult> {
+    const { type, icon } = headerFooter;
+    let config: LovelaceHeaderFooterConfig = { type };
+
+    if (this.hass) {
+      config = await getHeaderFooterStubConfig(
```

Comment:
We don't use this in the overview, so we should only do this when the user picks a header/footer.

We should add entities in a future PR, so we can render them and add support for custom cards probably.

Reply:
Maybe. I like the simplistic icons as well :)